packages/block-editor/src/components/use-setting/index.js

```diff
@@ -109,7 +109,7 @@ const removeCustomPrefixes = ( path ) => {
 export default function useSetting( path ) {
 	const { name: blockName, clientId } = useBlockEditContext();
 
-	const setting = useSelect(
+	return useSelect(
 		( select ) => {
 			if ( blockedPaths.includes( path ) ) {
```

Comment:
To me there is a discrepancy between the condition (`blockedPaths`) and the console warning in the consequence ("top-level paths"). Judging by the warning, I would ask: is the intention to block _any_ top-level path, or just these four (`[ 'color', 'border', 'typography', 'spacing' ]`)? If the former, could we rewrite the condition to:

```js
if ( path.includes( '.' ) ) {
```

Reply:
There are certain settings that can be accessed by its top-level path, such as `layout` (current usage in the codebase) and `border` (didn't find usage, but we could theoretically do that).